src/main/java/seedu/address/model/person/fields/subfields/NusMod.java

```diff
@@ -1,19 +1,45 @@
 package seedu.address.model.person.fields.subfields;
 
+import java.io.*;
+
 /**
  * Represents a Person's singular mod taken in the address book.
  */
 public class NusMod {
 
     public static final String MESSAGE_CONSTRAINTS = "Modules should be a part of NUS' NUSMods list";
+    public static final String MODULELIST_FILE_PATH = "data/moduleList.json";
     public final String name;
 
     public NusMod(String modsString) {
         this.name = modsString;
     }
 
-    //todo: Update modules to only be able to include mods that are a part of NUSMods.
+    /**
+     * Checks whether a module is a valid NUSMod by comapring the String name to the module list data
+     *
+     * @param trimmedTag The module name to be checked
+     * @return false if the module does not exist in NUSMods
+     */
     public static boolean isValidModName(String trimmedTag) {
-        return true;
+        try {
+            FileInputStream inputStream = new FileInputStream(MODULELIST_FILE_PATH);
```

Comment:
I think we will want to save it as a hashmap in the future so that we can very easily get the description of a module or name etc. Other than that it LGTM for now!